presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileReader.java

```diff
@@ -0,0 +1,613 @@
+            Slice valueClassName = readLengthPrefixedString(input);
+            if (!RCFILE_KEY_BUFFER_NAME.equals(keyClassName) || !RCFILE_VALUE_BUFFER_NAME.equals(valueClassName)) {
+                throw corrupt("File %s is a SequenceFile not an RCFile", dataSource);
+            }
+
+            compressed = input.readBoolean();
+
+            // RC file is never block compressed
+            if (input.readBoolean()) {
+                throw corrupt("File %s is a SequenceFile not an RCFile", dataSource);
+            }
+        }
+        else {
+            throw corrupt("File %s is not an RCFile", dataSource);
+        }
+
+        // setup the compression codec
+        if (compressed) {
+            Slice codecClassName = readLengthPrefixedString(input);
+            this.decompressor = codecFactory.createDecompressor(codecClassName.toStringUtf8());
+        }
+        else {
+            this.decompressor = null;
+        }
+
+        // read metadata
+        int metadataEntries = Integer.reverseBytes(input.readInt());
+        if (metadataEntries < 0) {
+            throw corrupt("Invalid metadata entry count %s in RCFile %s", metadataEntries, dataSource);
+        }
+        if (metadataEntries > MAX_METADATA_ENTRIES) {
+            throw corrupt("Too many metadata entries (%s) in RCFile %s", metadataEntries, dataSource);
+        }
+        ImmutableMap.Builder<String, String> metadataBuilder = ImmutableMap.builder();
+        for (int i = 0; i < metadataEntries; i++) {
+            metadataBuilder.put(readLengthPrefixedString(input).toStringUtf8(), readLengthPrefixedString(input).toStringUtf8());
+        }
+        metadata = metadataBuilder.build();
+
+        // get column count from metadata
+        String columnCountString = metadata.get(COLUMN_COUNT_METADATA_KEY);
+        try {
+            columnCount = Integer.parseInt(columnCountString);
+        }
+        catch (NumberFormatException e) {
+            throw corrupt("Invalid column count %s in RCFile %s", columnCountString, dataSource);
+        }
+
+        // initialize columns
+        if (columnCount > MAX_COLUMN_COUNT) {
+            throw corrupt("Too many columns (%s) in RCFile %s", columnCountString, dataSource);
+        }
+        columns = new Column[columnCount];
+        for (int columnIndex = 0; columnIndex < columns.length; columnIndex++) {
+            ColumnEncoding columnEncoding = encoding.getEncoding(types.get(columnIndex));
+            columns[columnIndex] = new Column(columnEncoding, decompressor);
+        }
+
+        // read sync bytes
+        syncFirst = input.readLong();
+        syncSecond = input.readLong();
+
+        // seek to first row group
+        if (offset > input.position()) {
+            seekToFirstRowGroupInRange(offset, length);
+        }
+        // if the specified file region does not contain the start of a sync sequence, advance() will close the file on the first call
```

Comment:
Change this comment so that it covers the case where the first row group does not have a sync sequence.